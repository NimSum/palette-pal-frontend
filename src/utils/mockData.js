const mockData = {
  mockDetailedProjects: [
    {
      "project_name": "Project ONE",
      "project_id": 1,
      "palette_name": "Palette ONE",
      "palette_id": 1,
      "color_1": "#000000",
      "color_2": "#000000",
      "color_3": "#319547",
      "color_4": "#000000",
      "color_5": "#834267"
    },
    {
      "project_name": "Project TWO",
      "project_id": 2,
      "palette_name": "Palette TWO",
      "palette_id": 2,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267"
    },
    {
      "project_name": "Project ONE",
      "project_id": 2,
      "palette_name": "Palette THREE",
      "palette_id": 3,
      "color_1": "#123456",
      "color_2": "#654321",
      "color_3": "#245216",
      "color_4": "#784895",
      "color_5": "#235686"
    }
  ],
  mockProjects: [
    {
      "id": 1,
      "project_name": "Project ONE",
      "created_at": "2019-07-05T21:39:51.436Z",
      "updated_at": "2019-07-05T21:39:51.436Z"
    },
    {
      "id": 2,
      "project_name": "Project TWO",
      "created_at": "2019-07-05T21:40:51.436Z",
      "updated_at": "2019-07-05T21:40:51.436Z"
    }
  ],
  mockPalettes: [
    {
      "id": 1,
      "palette_name": "Test ONE",
      "project_id": 1,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267",
      "created_at": "2019-07-05T21:39:57.227Z",
      "updated_at": "2019-07-05T21:39:57.227Z"
    },
    {
        "id": 2,
        "palette_name": "Test TWO",
        "project_id": 2,
        "color_1": "#000000",
        "color_2": "#000000",
        "color_3": "#319547",
        "color_4": "#000000",
        "color_5": "#834267",
        "created_at": "2019-07-06T01:24:45.881Z",
        "updated_at": "2019-07-06T01:24:45.881Z"
    }
  ]
}

export default mockData;