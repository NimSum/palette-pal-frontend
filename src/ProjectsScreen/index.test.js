import React from 'react';
import ProjectScreen from './index';
import { shallow } from 'enzyme';
import mockData from '../utils/mockData';

const mockUpdatePaletteData = jest.fn();
const mockUpdateProjectData = jest.fn();

describe('ProjectScreen', () => {
  let wrapper;
  let instance;

  beforeEach(() => {
    wrapper = shallow(< ProjectScreen
      data={[mockData.mockCleanedProject]}
      updateProjectData={mockUpdateProjectData}
      updatePaletteData={mockUpdatePaletteData}
    />);

    instance = wrapper.instance();

    jest.spyOn(instance, 'closeDialog');
    jest.spyOn(instance, 'showDialog');
    jest.spyOn(instance, 'setProjectOption');
  })
  
  it('should match component ProjectScreen snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the expected default state', () => {
    expect(wrapper.state()).toEqual({
      showDialog: false,
      filter: '',
      format: 'hex',
      project: ''
    })
  })

  it('should set the showDialog property in state to true when showDialog is called', () => {
    expect(wrapper.state('showDialog')).toEqual(false);

    instance.showDialog();

    expect(wrapper.state('showDialog')).toEqual(true);
  })

  it('should set the showDialog property in state to false when closeDialog is called', () => {
    instance.showDialog();
    expect(wrapper.state('showDialog')).toEqual(true);

    instance.closeDialog();

    expect(wrapper.state('showDialog')).toEqual(false);
  })

  it('should set the project property in state to false when setProjectOption is called', () => {
    expect(wrapper.state('project')).toEqual('');

    instance.setProjectOption('test project');

    expect(wrapper.state('project')).toEqual('test project');
  })
})