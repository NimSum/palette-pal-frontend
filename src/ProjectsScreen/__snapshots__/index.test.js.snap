// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectScreen shoud match component ProjectScreen snapshot 1`] = `ShallowWrapper {}`;
